src/core/Rule.ml

```diff
@@ -906,15 +909,15 @@ let () = Printexc.register_printer opt_string_of_exn
 let visit_new_formula f formula =
```

Comment:
When manipulating formulas, I would use a different letter than 'f' to denote a "function", it's good that in a given context `f` is unambiguous. Maybe `func`?